musig2test.c

```diff
@@ -40,114 +26,204 @@ int main(void) {
     printf("* Number of messages\t: %d \n", NR_MSGS);
     printf("--------------------------------------------------------------------------- \n");
 
+    /* musig2test parameters. */
+    int i,j,k,l;
+    int ind;
+    secp256k1_pubkey pk_list[N];    // Signers' public key list
+    secp256k1_pubkey batch_list[N*V*NR_MSGS];   // Stores the batches of signers
+    unsigned char *parsig_list[N];  // The list of partial signatures
 
 
-    /* Create a keypair for each signer ******************************/
+    musig2_context_sig *mcs_list[N]; // Array that holds N musig2_context_sig
+
+    /**** Initialization ****/
     printf("\n-------- Initialize Signers ----------------------------------------------- \n");
-    for(i=0; i<N; i++){
-        signer_LIST[i] = malloc(sizeof (SIGNER));
-        signer_LIST[i]->keypair = malloc(sizeof (secp256k1_keypair));
-        MuSig2_KeyGen(ctx, signer_LIST[i]->keypair);
+    for (i=0; i<N; i++){
+        mcs_list[i] = malloc(sizeof (musig2_context_sig));
+        mcs_list[i]->mc = malloc(sizeof (musig2_context));
+        mcs_list[i]->mc->ctx = secp256k1_context_clone(ctx);
+
+        /* Generate a keypair for the signer and get batch commitments. */
+        musig2_init_signer(mcs_list[i]);
+
+        /* Store the public key of the signer in pk_list */
+        assert (secp256k1_keypair_pub(ctx, &pk_list[i], &mcs_list[i]->keypair));
+        l=0;
+
+        /* Store the batch commitments of the signer in batch_list */
+        for(k=0; k<NR_MSGS; k++){
+            for(j=0; j<V; j++,l++) {
```

Comment:
It doesn't have the same value as `j`. 

`j` is set to zero in each iteration of `for(k=0; k<NR_MSGS; k++)`, but `l` is the index of the signer's commitment list which is set to zero for each signer.  I used that system to handle collecting the batch commitments. But, I try to find a way to get rid of `l`,

Reply:
aha, you are right. I misread that. What about using the variables `k` and `j` instead? so change `l` by `k * V + j`. Or otherwise you can add a little comment to what `l` is. whatever you prefer.